ujt/utils.py

```diff
@@ -11,54 +11,56 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-""" Temporary main class for prototyping purposes.
+""" Module providing utility functions.
 
-Currently used for testing protobuf read and write functionality.
+Currently used for protobuf read and write functionality.
 """
 
 import pathlib
-from typing import Any
+from typing import Type
 
 import google.protobuf.text_format as text_format
+from google.protobuf.message import Message
 
-from generated import graph_structures_pb2
 
+def file_name(name: str, proto_type: Type[Message]):
+    return f"{proto_type.__name__}_{name}"
 
-def write_proto_to_file(message: graph_structures_pb2.Service) -> None:
+
+def write_proto_to_file(message: Message, proto_type: Type[Message]) -> None:
     """Writes a protobuf to disk in a human-readable format.
 
     Args:
         message: A protobuf message with a name field.
+        proto_type: The type of the protobuf message.
 
     Raises:
         ValueError: The provided message's name field was empty.
         AttributeError: The provided message does not contain a name field.
     """
 
-    if not message.name:
+    if not message.name:  # type: ignore
```

Comment:
Mypy complains: `ujt/utils.py:42: error: "Message" has no attribute "name"`
This can be fixed by specifying the type of message more precisely (e.g. Union[Service, Client, ...]). Should we explicitly specify these types? It seems too restrictive, as ideally the method should support writing protobufs with _any_ message type with a name field. (We also specify that it could raise AttributeError in the docstring.)

Reply:
I think its better to write a generic message. Are there really no field existence checks in this language? Perhaps alternatively can have the name supplied as a method parameter as well.

Reply:
It's possible to check if a variable has a field with [hasattr](https://hynek.me/articles/hasattr/), but Python style prefers asking forgiveness rather than permission (i.e. try to access the field and if it doesn't exist, catch the error). 

At runtime, Python will raise an attribute error saying message has no attribute name.
Mypy (a static type checker) is not satisfied with the type definitions here.

I'll probably just keep suppressing the Mypy error but provide some additional context by explicitly raising AttributeError in the code. Thanks!